core/src/main/java/com/yahoo/ycsb/measurements/PeriodicHistogram.java

```diff
@@ -0,0 +1,352 @@
+  public static final String BUCKET_INTERVAL_DEFAULT = "1.0";
+
+  public static final String OUTPUT_DIR = "periodichistogram.output.path";
+  public static final String OUTPUT_DIR_DEFAULT = "";
+  
+  private static String NEWLINE = System.getProperty("line.separator");
+  
+  /**
+   * Specify the range of latencies to track in the histogram.
+   */
+  final int buckets;
+	
+  /**
+   * Groups operations in discrete blocks of 1ms width.
+   */
+  int[] histogram;
+  
+  /**
+   * Histogram of previous status thread iteration.
+   */
+  int[] prevhistogram;
+  
+  /**
+   * Counts all operations outside the histogram's range.
+   */
+  int histogramoverflow;
+  
+  /**
+   * Overflow of previous status thread iteration.
+   */
+  int prevhistogramoverflow;
+  
+  /**
+   * The total number of reported operations.
+   */
+  int operations;
+  
+  /**
+   * Reported operations from previous status thread iteration.
+   */
+  int prevoperations;
+  
+  /**
+   * The latency bucket multiplier used to convert latency in microseconds to the proper bucket index.
+   */
+  double latencymultiplier;
+
+  /**
+   * The sum of each latency measurement over all operations.
+   * Calculated in ms.
+   */
+  long totallatency;
+
+  /**
+   * The sum of each latency measurement squared over all operations. 
+   * Used to calculate variance of latency.
+   * Calculated in ms. 
+   */
+  double totalsquaredlatency;
+  
+  /**
+   * Start of status interval in milliseconds since epoch.
+   */
+  long windowstart;
+  
+  /**
+   * End of status interval in milliseconds since epoch.
+   */
+  long windowend;
+
+  /**
+   * Minimum latency for entire run.
+   */
+  int min;
+  
+  /**
+   * Maximum latency for entire run.
+   */
+  int max;
+  
+  /**
+   * Number of seconds in each bucket expressed as a string.  Used in json output.
+   */
+  final String secondsperbucketstring;
+
+  /**
+   * File path of periodic json histogram.
+   */
+  String path;
+  
+  /**
+   * File writer for periodic json histogram.
+   */
+  FileWriter writer;
+  
+  /**
+   * Reusable string builder.
+   */
+  StringBuilder sb;
+  
+  /**
+   * Floating point formatter.
+   */
+  DecimalFormat secondsFormat;
+  DecimalFormat opsecFormat;
+
+  public PeriodicHistogram(String name, Properties props)
+  {
+    super(name);
+
+    buckets = Integer.parseInt(props.getProperty(BUCKETS, BUCKETS_DEFAULT));
+    double bucketinterval = Double.parseDouble(props.getProperty(BUCKET_INTERVAL, BUCKET_INTERVAL_DEFAULT));
+    latencymultiplier = 1.0 / (bucketinterval * 1000.0);
+    secondsperbucketstring = new DecimalFormat("#.############").format(bucketinterval / 1000.0);
+
+    histogram = new int[buckets];
+    prevhistogram = new int[buckets];
+    histogramoverflow = 0;
+    prevhistogramoverflow = 0;
+    operations = 0;
+    prevoperations = 0;
+    totallatency = 0;
+    totalsquaredlatency = 0;
+    min = -1;
+    max = -1;
+    windowstart = windowend = System.currentTimeMillis();
```

Comment:
Sure about using System.cTM instead of System.nanoTime? the former will get moved around with the system clock